docs/connector-development/config-based/source_schema.yaml

```diff
@@ -122,73 +123,55 @@ definitions:
     enum:
       - GET
       - POST
-  RequestOptionsProvider:
-    type: object
-    anyOf:
-      - "$ref": "#/definitions/InterpolatedRequestOptionsProvider"
-  InterpolatedRequestOptionsProvider:
-    type: object
-    additionalProperties: true
-    properties:
-      "$options":
-        "$ref": "#/definitions/$options"
-      request_parameters:
-        "$ref": "#/definitions/RequestInput"
-      request_headers:
-        "$ref": "#/definitions/RequestInput"
-      request_body_data:
-        "$ref": "#/definitions/RequestInput"
-      request_body_json:
-        "$ref": "#/definitions/RequestInput"
   RequestInput:
     type: object
-    additionalProperties: true
+    additionalProperties: false
   Authenticator:
     type: object
     description: "Authenticator type"
     anyOf:
-      - "$ref": "#/definitions/OAuth"
+      - "$ref": "#/definitions/OAuthAuthenticator"
       - "$ref": "#/definitions/ApiKeyAuthenticator"
       - "$ref": "#/definitions/BearerAuthenticator"
       - "$ref": "#/definitions/BasicHttpAuthenticator"
   ApiKeyAuthenticator:
     type: object
-    additionalProperties: true
+    additionalProperties: false
     required:
       - header
-      - api_token
+      - api_key
     properties:
-      "$options":
-        "$ref": "#/definitions/$options"
-      header:
-        type: string
-      api_token:
+      "$parameters":
+        "$ref": "#/definitions/$parameters"
+      inject_into:
```

Comment:
it's technically a new feature for the ApiKeyAuthenticator to support request_parameters